crates/paimon/src/spec/types.rs

```diff
@@ -320,13 +323,16 @@ impl CharType {
 
     pub const MAX_LENGTH: usize = 255;
 
-    pub fn new(length: usize) -> Result<Self, &'static str> {
+    pub fn new(length: usize) -> Result<Self, Error> {
         Self::with_nullable(true, length)
     }
 
-    pub fn with_nullable(nullable: bool, length: usize) -> Result<Self, &'static str> {
+    pub fn with_nullable(nullable: bool, length: usize) -> Result<Self, Error> {
         if !(Self::MIN_LENGTH..=Self::MAX_LENGTH).contains(&length) {
-            return Err("Character string length must be between 1 and 255 (both inclusive).");
+            return DataTypeInvalidSnafu {
```

Comment:
Can we simplify this by `ensure!(length < Self::MIN_LENGTH, DataTypeInvalidSnafu {message: "Binary string length must be at least 1."});` ?

The same as below

Reply:
> Can we simplify this by `ensure!(length < Self::MIN_LENGTH, DataTypeInvalidSnafu {message: "Binary string length must be at least 1."});` ?
> 
> The same as below

This simplifies things a bit, but the readability of the code is reduced.

Reply:
Ok, make sense